test/test_multilayerqg.jl

```diff
@@ -431,9 +431,12 @@ function test_mqg_fluxes(dev::Device=CPU(); dt=0.001, stepper="ForwardEuler", n=
 
   ψ1 = @. cos(k₀*x) * cos(l₀*y) + sin(k₀*x)
   ψ2 = @. cos(k₀*x + π/10) * cos(l₀*y)
+  
   ψ = zeros(dev, eltype(gr), (gr.nx, gr.ny, nlayers))
-  CUDA.@allowscalar @views ψ[:, :, 1] .= ψ1
-  CUDA.@allowscalar @views ψ[:, :, 2] .= ψ2
```

Comment:
interesting... did you test whether `allowscalar` is still needed here? I belive this should use a `view` (even when not annotated with `@views`) but perhaps it doesn't work... there's something called `dotview` in Base that handles this case (broadcasting _into_ a slice)